src/pps/pps.proto

```diff
@@ -836,6 +839,26 @@ message RunCronRequest {
   Pipeline pipeline = 1 [(validate.rules).message.required = true];
 }
 
+// Request to check the status of pipelines within a project.
+message CheckStatusRequest {
+  oneof context {
+      // boolean field indicating status of all project pipelines. 
+      bool all = 1;
+      // project field
+      pfs_v2.Project project = 2;
+   }
+}
+
+// Response for check status request. Provides alerts if any.
+message CheckStatusResponse {
+  // project field
+  pfs_v2.Project project = 1;
```

Comment:
This info should be included in the `Pipeline` object, so I don't think you need it separated out here.